testing/tests/DevExpress.ui.widgets.form/form.tests.js

```diff
@@ -423,6 +424,76 @@ QUnit.test('From renders the right types of editors according to stylingMode opt
 });
 
+[true, false, undefined].forEach(repaintChangesOnly => {
+    [true, false].forEach(useRepaint => {
+        QUnit.test(`Form.itemOption('group.item1', newItem2), repaintChangesOnly = ${repaintChangesOnly}. useRepaint = ${useRepaint} (T903774)`, function(assert) {
+            const clone = (item) => JSON.parse(JSON.stringify(item));
+
+            const item1 = {
+                editorType: 'dxDropDownBox',
+                dataField: 'item1',
+                label: { text: 'item1' },
+                editorOptions: { placeholder: 'test_placeHolder' }
+            };
+            const newItem1 = {
+                editorType: 'dxTextBox',
+                dataField: 'newItem1',
+                label: { text: 'new item1' },
+                editorOptions: { width: 300 }
+            };
+            const form = $('#form').dxForm({
+                repaintChangesOnly,
+                items: [{
+                    itemType: 'group',
+                    caption: 'group1',
+                    items: [ clone(item1) ]
+                }]
+            }).dxForm('instance');
+
+            form.itemOption('group1.item1', clone(newItem1));
+            if(useRepaint) {
+                form.repaint();
+            }
+            if(repaintChangesOnly === false) {
+                assert.deepEqual(form.itemOption('group1.item1'), undefined, 'item1');
+                assert.deepEqual(form.itemOption('group1.newItem1'), newItem1, 'newItem1');
+            } else {
+                assert.deepEqual(form.itemOption('group1.item1'), extend(true, {}, item1, newItem1, { editorType: item1.editorType, dataField: item1.dataField }), 'item1');
+                assert.deepEqual(form.itemOption('group1.newItem1'), undefined, 'newItem1');
+            }
+        });
+
+        QUnit.test(`Form.itemOption('item1', newItem2), repaintChangesOnly = ${repaintChangesOnly}. useRepaint = ${useRepaint} (T903774)`, function(assert) {
+            const clone = (item) => JSON.parse(JSON.stringify(item));
+
+            const item1 = {
+                editorType: 'dxDropDownBox',
+                dataField: 'item1',
+                label: { text: 'item1' },
+                editorOptions: { placeholder: 'test_placeHolder' }
+            };
+            const newItem1 = {
+                editorType: 'dxTextBox',
+                dataField: 'newItem1',
+                label: { text: 'new item1' },
+                editorOptions: { width: 300 }
+            };
+            const form = $('#form').dxForm({
+                repaintChangesOnly,
+                items: [ clone(item1) ]
+            }).dxForm('instance');
+
+            form.itemOption('item1', clone(newItem1));
+            if(useRepaint) {
+                form.repaint();
+            }
+
+            assert.deepEqual(form.itemOption('item1'), undefined, 'item1');
+            assert.deepEqual(form.itemOption('newItem1'), newItem1, 'newItem1');
```

Comment:
inho it should not depend on the item' level and there should be the same "if(repaintChangesOnly === false) {"

Reply:
skip for new tickets